src/averageAges.js

```diff
@@ -36,8 +50,25 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
+
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const filteredTab = withChildren
+    ? people.filter(isMother)
+    : people.filter(isWoman);
+
+  function isWoman(person) {
+    return person.sex === 'f';
+  };
+
+  function isMother(person) {
+    return people.some(el => el.mother === person.name);
+  };
+
+  function sumAllAges(prev, next) {
```

Comment:
The `sumAllAges` function should be moved outside of `calculateWomenAverageAge` to keep the main function cleaner, and to avoid redefining it on each function call. 😇